lib/config/index.js

```diff
@@ -789,13 +789,16 @@ function initPrompts(newProject) {
 
     var curTranspiler = loader.transpiler;
 
+    // default transpiler will be typescript only if the main loader file ends with .ts
+    var defTranspiler = /\.ts$/.test(loader.package.main) ? 'typescript' : 'babel';
```

Comment:
`loader.package` an be null, so perhaps make this `.test(loader.package && loader.package.main)`.